src/modules/tools/zprobe/CalibrationStrategy.cpp

```diff
@@ -0,0 +1,597 @@
+                    repeats = int(v.second);
+                else
+                    parameters_to_optimize += v.first;
+            }
+
+            // TODO: the following checks are delta specific
+            if (parameters_to_optimize.empty()) {
+                parameters_to_optimize = "XYZ"; // endstop only
+            }
+            // The W parameter is linearly dependent on X,Y,Z
+            if (parameters_to_optimize.find('W') != std::string::npos &&
+                    (parameters_to_optimize.find('X') != std::string::npos ||
+                     parameters_to_optimize.find('Y') != std::string::npos ||
+                     parameters_to_optimize.find('Z') != std::string::npos)) {
+                gcode->stream->printf("Warning, parameter W is coupled to X,Y,Z. Optimize either W or XYZ. Ignoring W.\n");
+                parameters_to_optimize.erase(parameters_to_optimize.find('W'));
+            }
+
+            // Make sure we initialize our trim variables
+            get_parameter('X');
+            get_parameter('Y');
+            get_parameter('Z');
+
+            // Make sure all parameters are valid
+            for (auto p = parameters_to_optimize.begin(); p != parameters_to_optimize.end();) {
+                // this is done by attempting to get and then set their values
+                if (!update_parameter(*p,0)) {
+                    gcode->stream->printf("Warning, invalid parameter '%c' ignored.\n", *p);
+                    p = parameters_to_optimize.erase(p);
+                } else {
+                    p++;
+                }
+            }
+
+            update_compensation_transformation(); // make sure no compensation transform from somewhere else is installed
+
+            gcode->stream->printf("Commencing calibration of parameters: %s\n", parameters_to_optimize.c_str());
+
+            samples = std::max(samples, (int)parameters_to_optimize.length());
+
+            if (optimize_model(samples, repeats, parameters_to_optimize, gcode->stream)) {
+                gcode->stream->printf("Calibration complete. Save settings with M500.\n");
+            } else {
+                gcode->stream->printf("Calibration may not have converged. Use M500 if you want to save settings anyway.\n");
+            }
+            THEKERNEL->call_event(ON_IDLE);
+
+            // Check if endstop trim was updated
+            if (parameters_to_optimize.find('X') != std::string::npos ||
+                parameters_to_optimize.find('Y') != std::string::npos ||
+                parameters_to_optimize.find('Z') != std::string::npos) {
+
+                // Deficiency: homing is the only way to activate the endstop trim values
+                zprobe->home();
+            }
+
+            return true;
+        }
+    } else if(gcode->has_m) {
+        // handle mcodes
+        if(gcode->m == 500 || gcode->m == 503) { // M500 save, M503 display
+            gcode->stream->printf(";Plane tilt:\n");
```

Comment:
This should only be output if U V or W are set, otherwise it should not be set. It will waste config-override space if it is set when not used (ie on deltas).